urlb57eb64056_jacadenac_ISI_Othello_UN_tgz-pJ8-othello_OthelloJ8/src/othello/UIJuego.java

```diff
@@ -0,0 +1,427 @@
+		controlador.inicializar(10, 10, TipoTablero.CLASICO);
+		iniciarTableroUI();
+	}
+	
+	private void iniciarTableroUI(){		
+		this.estaModificandoTablero = false;
+		TipoTablero tipoTablero = controlador.obtenerTableroLogica().obtenerTipoTablero();
+		tableroUI = new TableroUI(Tablero.TABLERO_LARGO,Tablero.TABLERO_ANCHO, controlador.obtenerTableroLogica(), tipoTablero);
+		tableroUI.setVisible(true);
+		
+		tableroUI.obtenerItemNuevoJuego().addActionListener(new ActionListener() {
+			@Override
+			public void actionPerformed(ActionEvent e) {
+					definirTipoTablero(obtenerTamTableroElegido(), obtenerTipoTableroElegido());
+			}
+		});
+		
+		tableroUI.obtenerItemAbrirJuego().addActionListener(new ActionListener() {
+			@Override
+			public void actionPerformed(ActionEvent e) {				
+				JFileChooser fileChooser = new JFileChooser();
+				fileChooser.setCurrentDirectory(directorio);	
+				fileChooser.setFileFilter(miFiltro);
+				int seleccion = fileChooser.showOpenDialog(tableroUI);
+				if (seleccion == JFileChooser.APPROVE_OPTION)
+				{
+				   File fichero = fileChooser.getSelectedFile();
+				   try {
+					   tableroUI.dispose();
+					   leerArchivo(fichero, null);
+					   tableroUI = new TableroUI(Tablero.TABLERO_LARGO,Tablero.TABLERO_ANCHO, controlador.obtenerTableroLogica(), null);
+					   iniciarTableroUI();
+					   run();
+				   } catch (FileNotFoundException e1) {
+					   e1.printStackTrace();
+				   }
+				}
+			}
+		});
+		
+		tableroUI.obtenerItemGuardarJuego().addActionListener(new ActionListener() {
+			@Override
+			public void actionPerformed(ActionEvent e) {				
+				JFileChooser fileChooser = new JFileChooser();
+				directorio.mkdirs();
+				fileChooser.setCurrentDirectory(directorio);		
+				fileChooser.setFileFilter(miFiltro);
+				int seleccion = fileChooser.showSaveDialog(tableroUI);
+				if (seleccion == JFileChooser.APPROVE_OPTION)
+				{
+				   String path = fileChooser.getSelectedFile().getAbsolutePath();
+				   String extension =  miFiltro.getDescription().substring(miFiltro.getDescription().length()-4);
+				   if (!path.substring(path.length()-4).equals(extension)){
+					   path = path + extension;
+				   }
+				   File fichero = new File(path);
+				   escribirArchivo(fichero);			   
+				}
+			}
+		});
+
+		for (final JRadioButtonMenuItem tipoTableroButton : tableroUI.obtenerTipoTableroButton()) {
+			tipoTableroButton.addActionListener(new ActionListener() {
+
+				@Override
+				public void actionPerformed(ActionEvent e) {
+					definirTipoTablero(obtenerTamTableroElegido(), obtenerTipoTableroElegido());
+				}
+			});
+		}
+		
+		for (final JRadioButtonMenuItem tamTableroButton : tableroUI.obtenerTamTableroButton()) {
+			tamTableroButton.addActionListener(new ActionListener() {
+
+				@Override
+				public void actionPerformed(ActionEvent e) {
+					definirTipoTablero(obtenerTamTableroElegido(), obtenerTipoTableroElegido());
+				}
+			});
+		}		
+	}
+	
+	@Override
+	public void run() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.